bftengine/src/bcstatetransfer/RangeValidationTree.cpp

```diff
@@ -520,7 +529,14 @@ bool RangeValidationTree::validateTreeValues() const noexcept {
   return true;
 }
 
-bool RangeValidationTree::validate() const noexcept { return validateTreeStructure() && validateTreeValues(); }
+bool RangeValidationTree::validate() const noexcept {
+  auto ret1 = validateTreeStructure();
```

Comment:
That's risky, if validateTreeStructure is invalid, you dont want to validateTreeValues as you will crash. This can be a good way to crash us with byzantine source.
I suggest to write it with the same logic as before:

auto ret = validateTreeStructure();
if (!ret) {
 metrics_.rvt_validation_failures_++;
 return false;
}
ret = validateTreeValues();
if (!ret) {
 metrics_.rvt_validation_failures_++;
 return false;
}
return true;

Reply:
Thanks. Updated accordingly.